velox/dwio/common/SortingWriter.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/dwio/common/Writer.h"
+#include "velox/exec/SortBuffer.h"
+
+namespace facebook::velox::dwio::common {
+
+/// Sorting Writer object is used to write sorted data into a single file.
+class SortingWriter : public Writer {
+ public:
+  SortingWriter(
+      std::unique_ptr<exec::SortBuffer> sortBuffer,
+      std::unique_ptr<Writer> writer);
+
+  virtual void write(const RowVectorPtr& data) override;
+
+  /// No action because we need to accumulate all data and sort before data can
+  /// be flushed
+  virtual void flush() override;
+
+  virtual void close() override;
+
+  virtual void abort() override;
+
+  std::unique_ptr<exec::SortBuffer> sortBuffer_;
```

Comment:
```
const std::unique_ptr<exec::SortBuffer> sortBuffer_;
```

Reply:
We need to call reset on sortBuffer_ in `abort` function, so can't set it to const